src/Services/DataService/Epics/FetchRecordEpic.test.ts

```diff
@@ -0,0 +1,761 @@
+
+      initializeTestServices({}, { coalesceFindRequests: false });
+    });
+
+    describe("when coalesceFindRequests is true", () => {
+
+      const timeOutPeriod = DEFAULT_COALESCE_BUFFER_TIME + 1;
+
+      beforeEach(() => {
+        initializeTestServices({}, { coalesceFindRequests: true });
+      });
+
+      it("should emit a FETCH_ALL action from the first observable when called multiple times within the timeout period", () => {
+        const id = random.number().toString();
+        const secondId = random.number().toString();        
+
+        const epic = fetchRecordEpic(context);
+        const observableAction = ActionsObservable.of({
+          type: context.types.FETCH_RECORD,
+          payload: { id },
+        });
+        const sut = epic(observableAction, store);
+
+        let action;
+        sut
+          .take(1)
+          .subscribe(a => action = a);
+
+        const secondObservableAction = ActionsObservable.of({
+          type: context.types.FETCH_RECORD,
+          payload: { id: secondId },
+        });
+
+        epic(secondObservableAction, store)
+          .subscribe();
+
+        const queryBuilder = new QueryBuilder(context.name, { ids: [id, secondId] });
+        return new Promise((resolve, reject) => {
+          setTimeout(() => {
+            try {
+              expect(action).to.deep.equal({
+                type: "FETCH_ALL",
+                payload: queryBuilder,
+              });
+              resolve();
+            } catch (e) {
+              reject(e);
+            }
+          }, timeOutPeriod);
+        });
+      });
+
+      it("should emit a PUSH_RECORD action from the observable when called once", () => {
+        const id = random.number().toString();
+
+        const observableAction = ActionsObservable.of({
+          type: context.types.FETCH_RECORD,
+          payload: { id },
+        });
+
+        const epic = fetchRecordEpic(context);
+        const sut = epic(observableAction, store);
+
+        let action;
+        sut
+          .take(1)
+          .subscribe(a => action = a);
+
+        return new Promise((resolve, reject) => {
+          setTimeout(() => {
+            try {
+              expect(action).to.deep.equal({
+                type: "PUSH_RECORD",
+                payload: { 
+                  fullText: expectedResult.fullText,
+                  id, 
+                },
+              });
+              resolve();
+            } catch (e) {
+              reject(e);
+            }
+          }, timeOutPeriod);
+        });
+      });
+
+    });
+
+    describe("when coalesceFindRequests is false", () => {
+      const shortTimeout = DEFAULT_COALESCE_BUFFER_TIME / 2.0;
+      const promiseTimeout = DEFAULT_COALESCE_BUFFER_TIME + 2;
+
+      it("should not buffer the request", () => {
+        const id = random.number().toString();
+        const secondId = random.number().toString();
+        const sut = createBufferObservable(context);
```

Comment:
shouldn't this be testing the fetchRecordEpic, not createBufferObservable?

Reply:
Yeah, I think so.  It would be nice to be able to stub out `performBufferedRequest` and `loadRecord` so that we could just test these directly for whether they are called in `fetchRecordEpic`.  I like that we've broken apart these functions, but it didn't buy us much in the way of unit testing the individual components.